src/BuiltInTools/dotnet-watch/EnvironmentVariablesBuilder.cs

```diff
@@ -38,5 +38,19 @@ public void AddToEnvironment(Dictionary<string, string> variables)
             variables.Add(EnvironmentVariables.Names.DotnetStartupHooks, string.Join(s_startupHooksSeparator, DotNetStartupHooks));
             variables.Add(EnvironmentVariables.Names.AspNetCoreHostingStartupAssemblies, string.Join(AssembliesSeparator, AspNetCoreHostingStartupAssemblies));
         }
+
+        public IEnumerable<string> ToCommandLineDirectives()
+        {
+            foreach (var (name, value) in this)
+            {
+                yield return MakeDirective(name, value);
+            }
+
+            yield return MakeDirective(EnvironmentVariables.Names.DotnetStartupHooks, string.Join(s_startupHooksSeparator, DotNetStartupHooks));
+            yield return MakeDirective(EnvironmentVariables.Names.AspNetCoreHostingStartupAssemblies, string.Join(AssembliesSeparator, AspNetCoreHostingStartupAssemblies));
+
+            static string MakeDirective(string name, string value)
+                => $"[env:{name}={value}]";
```

Comment:
Could value reasonably contain an "="?

Reply:
Definitely. The parser finds the first `=` and treats the rest as the value: https://github.com/dotnet/command-line-api/blob/c96672b8b84c307feb035fed6cbe9db85d5b87d3/src/System.CommandLine/EnvironmentVariablesDirective.cs#L53

Reply:
That said, if the path contains space it breaks :(. We need to escape.